packages/dd-trace/test/opentracing/propagation/text_map.spec.js

```diff
@@ -523,6 +523,19 @@ describe('TextMapPropagator', () => {
       expect(spanContext._tracestate).to.be.undefined
     })
 
+    it('extracts span_id from tracecontext headers and stores datadog parent-id in trace_distributed_tags', () => {
```

Comment:
There was a lot of new code with many branches added. Should there be more tests for all combinations?

Reply:
All the branches should be tested by the following system tests: https://github.com/DataDog/system-tests/blob/main/tests/parametric/test_headers_tracecontext.py#L717-%23L809 (including the scenario in this unit test). 

We are trying to avoid writing the same test cases in 8 different languages.

Reply:
I understand the goal, but with that approach, how can we know that things continue to work while working locally? And how do we monitor code coverage?

Reply:
Yup we can add one more test and test should give us full code coverage. We can use the system tests to address edges with unexpected inputs.

Reply:
Added a new unit test to address all branch's